strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java

```diff
@@ -0,0 +1,29 @@
+package org.carlspring.strongbox.authentication.api.impl.ldap;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class LdapAuthenticator
```

Comment:
Right, but `2.` can be done with `ProviderManager`. Actually the `AuthenticationProvider` chain have already implemented there and we can be sure that it will work well as it's used by many people. Also it will be clearer for users who wish to implement own Authenticators to use standard Spring components for this. And it's integrated out of the box with other Spring components.